src/modules/launcher/PowerLauncher/ViewModel/MainViewModel.cs

```diff
@@ -524,6 +523,54 @@ private void QueryResults()
                                     Results.Visibility = Visibility.Hidden;
                                 }
                             }));
+
+                            // Run the slower query of the DelayedExecution plugins
+                            currentCancellationToken.ThrowIfCancellationRequested();
+                            foreach (PluginPair plugin in plugins)
+                            {
+                                if (!plugin.Metadata.Disabled && plugin.Metadata.DelayedExecution)
+                                {
+                                    var results = PluginManager.QueryForPlugin(plugin, query, true);
+                                    currentCancellationToken.ThrowIfCancellationRequested();
+                                    if (results.Count != 0)
+                                    {
+                                        lock (_addResultsLock)
+                                        {
+                                            if (query.RawQuery == _currentQuery.RawQuery)
+                                            {
+                                                currentCancellationToken.ThrowIfCancellationRequested();
+
+                                                // Remove the original results from the plugin
+                                                Results.Results.RemoveAll(r => r.Result.PluginID == plugin.Metadata.ID);
+                                                currentCancellationToken.ThrowIfCancellationRequested();
+
+                                                // Add the new results from the plugin
+                                                UpdateResultView(results, query, currentCancellationToken);
+                                                currentCancellationToken.ThrowIfCancellationRequested();
+                                                Results.Sort();
+                                            }
+                                        }
+
+                                        currentCancellationToken.ThrowIfCancellationRequested();
+                                        Application.Current.Dispatcher.BeginInvoke(new Action(() =>
```

Comment:
I moved the dispatcher lines of code to a function. Since the logic in the _addResultsLock blocks are slightly different I didnt move those into a function